pkg/component/component.go

```diff
@@ -531,6 +531,25 @@ func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo
 	if err = Update(client, componentConfig, componentConfig.GetSourceLocation(), stdout); err != nil {
 		return err
 	}
+	// URL creation part
+	urls := componentConfig.GetUrl()
```

Comment:
Can we move this section(url checks and creation) into a separate function and call the function here

Reply:
fixed